### PR TITLE
Feat(eos_cli_config_gen): add ip verify unicast source to ethernet-, vlan- and port-channel interfaces

### LINKED ISSUE #2880: Feat(eos_cli_config_gen): add ip verify unicast source to ethernet-, sub- and vlan-interfaces
### Enhancement summary

add unicast reverse path forwarding (uRPF) to ethernet- and sub-interfaces and Vlan interfaces:

```
leaf-test(config-if)#ip verify ?
  unicast  Enable RPF on unicast traffic

leaf-test(config-if)#ip verify unicast ?
  source  RPF mode

leaf-test(config-if)#ip verify unicast source ?
  reachable-via  RPF mode

leaf-test(config-if)#ip verify unicast source reachable-via ?
  any  Set RPF check to loose-mode
  rx   Set RPF check to strict-mode
  ```
  

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

```
vlan_interfaces:
  - name: Vlan1021
    ip_verify_unicast_source_reachable_via: "rx"
```

```
interface Vlan1021
  ip verify unicast source reachable-via rx
```

### Describe the solution you would like

```
ethernet_interfaces:
  - name: <str>
    ip_verify_unicast_source_reachable_via: < any|rx >
```

```
vlan_interfaces:
  - name: <str>
    ip_verify_unicast_source_reachable_via: < any|rx >
```

### Describe alternatives you have considered

usage of eos_cli

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Shivani-gslab**
## Change Summary

Add ip verify unicast source to ethernet-, vlan- and port-channel interfaces

## Related Issue(s)

Fixes #2880

## Component(s) name

`arista.avd.eos_cli_config_gen`

### User Checklist

- [x] Verify the ordering on EOS switch

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ethernet-interfaces.j2 [hunk 1]
```diff
@@ -341,6 +341,13 @@ interface {{ ethernet_interface.name }}
 {%     if ethernet_interface.ip_address is arista.avd.defined("dhcp") and ethernet_interface.dhcp_client_accept_default_route is arista.avd.defined(true) %}
    dhcp client accept default-route
 {%     endif %}
+{%     if ethernet_interface.ip_verify is arista.avd.defined %}
+{%         if ethernet_interface.ip_verify.unicast_source is arista.avd.defined %}
+{%             if ethernet_interface.ip_verify.unicast_source.reachable_via is arista.avd.defined %}
+   ip verify unicast source reachable-via {{ ethernet_interface.ip_verify.unicast_source.reachable_via }}
+{%             endif %}
+{%         endif %}
+{%     endif %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ethernet-interfaces.j2]: Same for the other places.
```suggestion
{%     if ethernet_interface.ip_verify.unicast_source.reachable_via is arista.avd.defined %}
   ip verify unicast source reachable-via {{ ethernet_interface.ip_verify.unicast_source.reachable_via }}
{%     endif %}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2 [hunk 2]
```diff
@@ -50,6 +47,16 @@ interface {{ vlan_interface.name }}
...
 {%         endif %}
 {%     endif %}
+{%     if vlan_interface.ip_verify is arista.avd.defined %}
+{%         if vlan_interface.ip_verify.unicast_source is arista.avd.defined %}
+{%             if vlan_interface.ip_verify.unicast_source.reachable_via is arista.avd.defined %}
+   ip verify unicast source reachable-via {{ vlan_interface.ip_verify.unicast_source.reachable_via }}
+{%             endif %}
+{%         endif %}
+{%     endif %}
+{%     if vlan_interface.ip_directed_broadcast is arista.avd.defined(true) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2]: unrelated change?

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/vlan-interfaces.md [hunk 3]
```diff
@@ -215,6 +218,9 @@
 
             # IPv4_address/Mask
           - <str>
+        ip_verify:
+          unicast_source:
+            reachable_via: <str; "any" | "rx">
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/vlan-interfaces.md]: On the device where I tested, I could not find any other commands through this cli. I would consider collapsing to a single key: `ip_verify_unicast_source_reachable_via: <str; "any | "rx">`
@carlbuchmann thoughts?

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2 (hunk 2 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2]: its for the sequence/ordering, I checked on EOS

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ethernet-interfaces.j2 (hunk 1 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ethernet-interfaces.j2]: Updated!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2 (hunk 2 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2]: We should not be changing ordering of existing commands. Just insert the new command the best you can and ignore the existing.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/vlan-interfaces.md (hunk 3 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/vlan-interfaces.md]: Please collapse to a single key `ip_verify_unicast_source_reachable_via` thanks.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2 (hunk 2 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2]: reverted it

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/vlan-interfaces.md (hunk 3 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/vlan-interfaces.md]: Fixed!

**Review by ClausHolbechArista — APPROVED** [review]

**Review by emilarista — APPROVED** [review]
Tested, LGTM!
